### DISCUSSION TITLE
teal.gallery use with my CDSIC data

### DISCUSSION THREAD

**Discussion by asjohans** [insightsengineering/teal.gallery#23] [Q&A]
I would like to use this app to look at my own CDISC data. I am a SAS and Python programmer and R is new to me. I see where the data is being imported from the SCDA.2022 library. Could I read in my data as shown below and comment out the other ADSL?  I also see one other place where the data seems to be imported. How should I handle that?  I mainly need to prove the concept and then have an R programmer take over to program it correctly.  Thanks for any help you can provide! 

#ADSL <- read_sas('D:\Users\xxx.xxxxx\Documents\r_code\test_teal\Test_data\adsl_dummy.sas7bdat')
ADSL <- synthetic_cdisc_data("latest")$adsl
ADRS <- synthetic_cdisc_data("latest")$adrs
ADLB <- synthetic_cdisc_data("latest")$adlb

adsl <- cdisc_dataset("ADSL", ADSL, code = "ADSL <- synthetic_cdisc_data(\"latest\")$adsl")
adrs <- cdisc_dataset("ADRS", ADRS, code = "ADRS <- synthetic_cdisc_data(\"latest\")$adrs")
adlb <- cdisc_dataset("ADLB", ADLB, code = "ADLB <- synthetic_cdisc_data(\"latest\")$adlb")

**Comment by donyunardi** [comment]
Hi @asjohans 

Yes, you should be able to reuse the `app.R` code in this package and plug in your own dataset.

Your update should be done before this line, because this is where `teal` app got initialize:
https://github.com/insightsengineering/teal.gallery/blob/00547871a0a0131857dc6938ea153edad3df2c5c/inst/apps/early-dev/app.R#L405

Your code should look something line this:
```r
# your data code and pre-processing, if applicable
ADSL <- read_sas("your-local-path")
adsl <- cdisc_data("ADSL", ADSL) # make sure to include `code` and `vars` argument if needed

# then init starts here:
app <- teal::init( 
  data = cdisc_data(
    adsl,
    ...
```

However, I would like to inform you in advance that certain teal modules require the data to be structured in a specific format for the analysis to function properly. As an illustration, one instance could be where a teal module anticipates a variable with factor levels. It is important to be aware of this requirement. If you encounter any issues with a particular column or variable, there should be a message prompting the issue at hand, or another approach would be to compare the data type in your dataset with the one generated by the data processing mentioned in the `app.R` file. 

Hope this helps.
